sdks/js/packages/core/react/components/organization/tokens/index.tsx

```diff
@@ -0,0 +1,141 @@
+import coin from '~/react/assets/coin.svg';
+
+interface TokenHeaderProps {
+  billingSupportEmail?: string;
+  isLoading?: boolean;
+}
+
+const TokensHeader = ({ billingSupportEmail, isLoading }: TokenHeaderProps) => {
+  return (
+    <Flex direction="column" gap="small">
+      {isLoading ? (
+        <Skeleton containerClassName={tokenStyles.flex1} />
+      ) : (
+        <Text size={6}>Tokens</Text>
+      )}
+      {isLoading ? (
+        <Skeleton containerClassName={tokenStyles.flex1} />
+      ) : (
+        <Text size={4} style={{ color: 'var(--foreground-muted)' }}>
+          Oversee your billing and invoices.
+          {billingSupportEmail ? (
+            <>
+              {' '}
+              For more details, contact{' '}
+              <a
+                href={`mailto:${billingSupportEmail}`}
+                target="_blank"
+                style={{ fontWeight: 400, color: 'var(--foreground-accent)' }}
+              >
+                {billingSupportEmail}
+              </a>
+            </>
+          ) : null}
+        </Text>
+      )}
+    </Flex>
+  );
+};
+
+function getFormatedNumberString(num: Number = 0) {
+  const numString = num.toString();
+  const length = numString.length;
+
+  return numString.split('').reduce((acc, val, i) => {
+    const diff = length - i;
+    if (diff % 3 === 0 && diff < length) {
+      return acc + ',' + val;
+    }
+    return acc + val;
+  }, '');
+}
+
+interface BalancePanelProps {
+  balance: number;
+  isLoading: boolean;
+}
+
+function BalancePanel({ balance, isLoading }: BalancePanelProps) {
+  const formatedBalance = getFormatedNumberString(balance);
+  return (
+    <Flex className={tokenStyles.balancePanel}>
+      <Flex className={tokenStyles.balanceTokenBox}>
+        {/* @ts-ignore */}
+        <Image src={coin} alt="coin" className={tokenStyles.coinIcon} />
+        <Flex direction={'column'} gap={'extra-small'}>
+          <Text weight={500} style={{ color: 'var(--foreground-muted)' }}>
+            Available tokens
+          </Text>
+          {isLoading ? (
+            <Skeleton style={{ height: '24px' }} />
+          ) : (
+            <Text size={9} weight={600}>
+              {formatedBalance}
+            </Text>
+          )}
+        </Flex>
+      </Flex>
+    </Flex>
+  );
+}
+
+export default function Tokens() {
+  const {
+    config,
+    client,
+    activeOrganization,
+    billingAccount,
+    isActiveOrganizationLoading,
+    isBillingAccountLoading
+  } = useFrontier();
+  const [tokenBalance, setTokenBalance] = useState(0);
+  const [isTokensLoading, setIsTokensLoading] = useState(false);
+
+  useEffect(() => {
+    async function getBalance(orgId: string, billingAccountId: string) {
+      try {
+        setIsTokensLoading(true);
+        const resp = await client?.frontierServiceGetBillingBalance(
+          orgId,
+          billingAccountId
+        );
+        const tokens = resp?.data?.balance?.amount || '0';
```

Comment:
In case token fetch fails, or we do not receive the amount, should we show an error message there instead of `0` ?